src/features/query-bar/components/filter/ValueField.js

```diff
@@ -2,32 +2,23 @@
 import * as React from 'react';
 import { FormattedMessage } from 'react-intl';
 
+import DatePicker from '../../../../components/date-picker';
 import SingleSelectField from '../../../../components/select-field/SingleSelectField';
 import TextInput from '../../../../components/text-input';
-import DatePicker from '../../../../components/date-picker';
-
-import messages from '../../messages';
+import type { OptionType } from '../../flowTypes';
 import { VALUE } from '../../constants';
+import messages from '../../messages';
 
 import '../../styles/Condition.scss';
 
 type Props = {
+    onChange: (option: SyntheticEvent<HTMLInputElement> | OptionType | Date) => void,
```

Comment:
I don't think ValueField should expose a function with such a complex signature. Is it possible to hide this complexity within ValueField e.g. onChange is invoked with `value`. This would require moving most of the complexity of `handleValueChange` (above) into ValueField.